commons-vfs2/src/main/java/org/apache/commons/vfs2/impl/DefaultFileSystemManager.java

```diff
@@ -704,9 +707,19 @@ public FileObject resolveFile(final FileObject baseFile, final String uri,
 
         // Extract the scheme
         final String scheme = UriParser.extractScheme(uri);
+        final Map<String,String> queryParam = UriParser.extractQueryParams(uri);
         if (scheme != null) {
             // An absolute URI - locate the provider
             final FileProvider provider = providers.get(scheme);
+            //In the case of SFTP set the path from root if the param is presented in URL
+            if (provider instanceof SftpFileProvider && "true".equals(queryParam.get(SftpConstants
+                    .SFTP_PATH_FROM_ROOT))) {
+                if (fileSystemOptions == null) {
+                    fileSystemOptions = new FileSystemOptions();
+                }
+                ((SftpFileSystemConfigBuilder) ((provider).getConfigBuilder())).setUserDirIsRoot
+                        (fileSystemOptions, false);
+            }
```

Comment:
In the case of SFTP what should happen if the param is NOT presented in URL

Reply:
"sftpPathFromRoot=true" needs to be added to the url as a query param if absolute paths are used
e.g.
<parameter name="transport.vfs.FileURI">vfs:sftp://user:pw@domain/home/user/test/in?vfs.passive=true%26sftpPathFromRoot=true</parameter>
So in this case set the path from root if this param is presented in URL by setting setUserDirIsRoot to false.
If this pram not presented assume user dir is the root.